### PR TITLE
Fix the argument order and add a test

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: test/script_tools.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 from vsc.install.testing import TestCase
 from vsc.utils.script_tools import ExtendedSimpleOption, DEFAULT_OPTIONS, CLI
 
+import vsc.utils.nagios
```
Comment (stdweird) [inline test/script_tools.py:39]: why is this needed?

File: test/script_tools.py [hunk 2]
```diff
@@ -146,3 +147,15 @@ def test_opts(self, prol):
...
+    @mock.patch('vsc.utils.nagios.ok_exit')
+    @mock.patch('vsc.utils.nagios.warning_exit')
+    @mock.patch('vsc.utils.nagios.unknown_exit')
+    @mock.patch('vsc.utils.nagios.critical_exit')
+    def test_exit(self, mock_critical, mock_unknown, mock_warning, mock_ok, mock_prologue):
+
+        cli = MyCLI()
+
+        cli.warning("be warned")
+        #mock_warning.is_called()
```
Comment (stdweird) [inline test/script_tools.py:160]: why don;t you mock `exit_from_errorcode` and see if it is called correctly?

File: test/script_tools.py (hunk 1 above)
Comment (itkovian) [reply test/script_tools.py:39]: It's not, it was a futile attempt to get mock to find the right code.

File: test/script_tools.py (hunk 2 above)
Comment (itkovian) [reply test/script_tools.py:160]: Because that would actually mean that I need to reproduce what it looks like here, which kinda circumvents the test. I want/need to just be able to call it.

Comment (itkovian) [reply test/script_tools.py:160]: If I would check how it is called, I can just call it in the right way :-p

Comment (stdweird) [reply test/script_tools.py:160]: the only thing you should test here if is you use the api correctly (looking at the commit, it was wrong before). if want to test the api, add tests for `vsc.utils.nagios`.

Comment (itkovian) [reply test/script_tools.py:160]: No, I called `exit_from_errorcode` in the wrong way. How would testing the nagios.py API help with that?

Comment (stdweird) [reply test/script_tools.py:160]: not at all. i'm only suggesting that you should only test that it is called correctly. checking if warning is called, only tests if exit_from_errorcode calls warning correctly if you call exit_from_errorcode correctly.

Comment (itkovian) [reply test/script_tools.py]: If it is incorrectly called, it will result in an unknown exit, hence have the wrong second argument.
